requirements-testing.txt

```diff
@@ -5,6 +5,8 @@ flake8==4.0.1
 pytest==6.2.5
 pytest-cov==3.0.0
 pytest-xdist == 2.4.0
+pytest-mock==3.6.1
```

Comment:
I ran into another scenario where I had to add `pytest-mock==3.6.1` within `tox.ini` lint section.  Can't we avoid doing it?  Unsure why we are not installing `requirements-testing` in there, which would have got this in automatically.  Had to go through CI to figure this out.

Reply:
Because the venv is cached and tox is not always able to figure out if the deps changed. You can add `-r` to tox when you want to force it to recreate a venv based on the new deps. Or you could `rm -rf .tox` (or some of its venvs) if you want.

Reply:
Oh wait, you're asking about `lint`. I guess that's because flake8 doesn't really need all of the deps listed here. OTOH we could distill a subset of the deps in a separate file that could indeed be reused.
The current dependencies setup is quite a mess so I skipped solving this problem for now. Ideally, a pip-tools managed structure should be implemented for proper separation of concerns.

Reply:
> OTOH we could distill a subset of the deps in a separate file that could indeed be reused.

Makes sense

> The current dependencies setup is quite a mess so I skipped solving this problem for now.

Totally.  I was thinking do we still need `requirements-* and simply not put them within `tox.ini`.  What will be the side-effects?

Reply:
> Ideally, a pip-tools managed structure should be implemented for proper separation of concerns.

Will look into it, haven't used it yet.

Reply:
I think it should be a separate effort, this PR is messy enough.

Reply:
> Totally. I was thinking do we still need `requirements-*` and simply not put them within `tox.ini`. What will be the side-effects?

Missed this part. Don't do this. I know that it's tempting but each tox env has a single list of deps while different runtime envs (OS/Python) would need separate lists of pins. Plus no tools are able to auto-update `tox.ini` anyway.
Oh, and once there's lockfiles/constaints with the exact pins, it'll be unsustainable to maintain hundreds of autogenerated lines in the middle of the tox config.